subprojects/core-api/src/test/groovy/org/gradle/api/tasks/util/PatternSetTest.groovy

```diff
@@ -16,18 +16,20 @@
 
 package org.gradle.api.tasks.util
 
+import groovy.transform.CompileStatic
 import org.apache.tools.ant.DirectoryScanner
 import org.gradle.api.file.FileTreeElement
 import org.gradle.api.file.RelativePath
 import org.gradle.api.specs.Spec
+import org.gradle.api.tasks.util.internal.PatternSpecFactory
 import spock.lang.Issue
 
 import static org.gradle.util.Matchers.strictlyEquals
 import static org.hamcrest.CoreMatchers.equalTo
 import static org.hamcrest.MatcherAssert.assertThat
 
 class PatternSetTest extends AbstractTestForPatternSet {
-    PatternSet patternSet = new PatternSet()
+    PatternSet patternSet = new TestPatternSet()
```

Comment:
I think this test should be fixed differently:
- Every time the test updates `DirectoryScanner.defaultExcludes`, it also needs to call `PatternSpecFactory.INSTANCE.setDefaultExcludesFromSettings(DirectoryScanner.getDefaultExcludes())`
- The cleanup also need to call `setDefaultExcludesFromSettings()` after resetting the default excludes
- There needs to be a new unit test which shows that using the spec fails when the default excludes changed without calling `setDefaultExcludesFromSettings()`.